### PR TITLE
Define status as struct

### PR DESCRIPTION
**Pull request by SimonLab**
use Struct to define the list of statuses, remove json file and `File.read` function
ref: #4 

@nelsonic let me know your thought on this,thanks

### CODE REVIEW

File: lib/status.ex [hunk 1]
```diff
@@ -0,0 +1,10 @@
+defmodule Status do
+  @moduledoc """
+  Define the Status struct with desc, id and text key required
+  """
+  @type t :: %Status{code: integer(), desc: String.t(), text: atom()}
+  @enforce_keys [:code, :desc, :text]
+  @derive Jason.Encoder
```
Comment (SimonLab) [inline lib/status.ex:7]: > If you need to encode some struct that does not implement the protocol, if you own the struct, you can derive the implementation specifying which fields should be encoded to JSON

https://github.com/michalmuskala/jason#differences-to-poison

File: lib/status.ex [hunk 2]
```diff
@@ -0,0 +1,10 @@
+defmodule Status do
+  @moduledoc """
+  Define the Status struct with desc, id and text key required
+  """
+  @type t :: %Status{code: integer(), desc: String.t(), text: atom()}
+  @enforce_keys [:code, :desc, :text]
```
Comment (SimonLab) [inline lib/status.ex:6]: all the keys are required

File: lib/statuses.ex [hunk 3]
```diff
@@ -1,47 +1,102 @@
...
+        desc: "an unexpected condition was encountered",
+        text: :internal_server_error
+      }
+    ]
   end
+
+  @doc """
+  Returns the list of statuses as json
+  """
+  def statuses_to_json(), do: Jason.encode!(get_statuses())
```
Comment (SimonLab) [inline lib/statuses.ex:101]: Create a function to return json, I don't think we'll use it but I can see how it can be useful

**Review by nelsonic — APPROVED** [review]
@SimonLab looks good. 👍 

Detailed response in the issue: https://github.com/dwyl/statuses/issues/4#issuecomment-1179609113

**Comment by nelsonic** [comment]
Package published to https://hex.pm/packages/statuses/1.1.0 📦 🚀

**Pull request merged**